src/components/Table/index.tsx

```diff
@@ -397,15 +397,33 @@ class Table<T> extends React.Component<TableProps<T>, TableState<T>> {
 	};
 
 	public resetPager = () => {
-		this.setState({ page: 0 });
+		const page = 0;
+
+		if (this.props.updatePage) {
```

Comment:
Let's extract this and the state setting in a `pageChange` method, and then each of these methods (reset, increment, decrement) will call that one with (0, page + 1, page - 1).